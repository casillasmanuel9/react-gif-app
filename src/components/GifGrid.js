import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: imagenes, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className={'animate__animated animate__fadeIn'}> {category} </h3>

      {loading && <p className={'animate__animated animate__flash'}> Loading.. </p>}

      <div className={"card-grid"}>
        {imagenes.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}