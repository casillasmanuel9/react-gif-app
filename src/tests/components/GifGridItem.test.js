import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import "@testing-library/jest-dom";

describe("Pruebas en componente", () => {
  const titulo = "Un titulo";
  const url = "https://localhost/img.gif";

  const wrapper = shallow(<GifGridItem title={titulo} url={url} />);

  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener el parrafo con el titulo", () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(titulo);
  });

  test("Debe de tener la imagen igual al url y alt de las props", () => {
    const img = wrapper.find('img');
    expect(img.props()['src']).toBe(url);
    expect(img.props()['alt']).toBe(titulo);
  });

  test('debe tener animated_fadeIn', () => {
      const div = wrapper.find('div');
      const className = div.prop('className');
      expect(className.includes('animate__fadeIn')).toBe(true);
  })
  
});
