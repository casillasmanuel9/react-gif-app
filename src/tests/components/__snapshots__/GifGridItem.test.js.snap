// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/img.gif"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
